### PR TITLE
Add RequiredRadioBooleanField and use it by default for required boolean columns

### PR DESCRIPTION
**Pull request by 3noch**

### CODE REVIEW

**Comment by 3noch** [comment]
@rsyring @nZac Can you guys take a look?

File: requirements/runtime.txt [hunk 1]
```diff
@@ -3,5 +3,6 @@
 
 https://github.com/level12/keg/zipball/master
 arrow
+wtforms_components
```
Comment (rsyring) [inline requirements/runtime.txt]: 1. Unless you have a direct dependency on this in our project, it should not be listed. It's a dependency of WTForms-Alchemy.
2. You didn't specify it correctly.  If it is needed here, it should be: WTForms-Components.

File: keg_elements/forms/__init__.py [hunk 2]
```diff
@@ -200,6 +201,26 @@ def selected_choice_label(self):
         return value_dict.get(self.data)
 
 
+class RequiredRadioBooleanField(wtforms.fields.RadioField):
+    def __init__(self, *args, **kwargs):
+        yes_label = kwargs.pop('yes_label', 'Yes')
+        no_label = kwargs.pop('no_label', 'No')
```
Comment (rsyring) [inline keg_elements/forms/__init__.py]: Yes/no labels should be "true_label" and "false_label".  This is a boolean field and internally, we should be dealing with true/false and have default labels of "yes"/"no".

File: keg_elements/forms/__init__.py [hunk 3]
```diff
@@ -200,6 +201,26 @@ def selected_choice_label(self):
...
+        def bool_coerce(val):
+            if val == u'True':
+                return True
+            if val == u'False':
+                return False
+            return val
+
+        kwargs['choices'] = [(True, yes_label), (False, no_label)]
+        kwargs['coerce'] = bool_coerce
+        kwargs['validators'] = [InputRequired()]
```
Comment (rsyring) [inline keg_elements/forms/__init__.py]: What if we want other validators to be able to be used.  By not adding to the values that may be in kwargs you eliminate the ability to send in other validators.

File: keg_elements/forms/__init__.py [hunk 4]
```diff
@@ -12,6 +12,7 @@
 import six
 import wtforms.fields
 import wtforms.form
+from wtforms import widgets
```
Comment (rsyring) [inline keg_elements/forms/__init__.py]: Is this used somewhere?  I don't see it in the diff.

**Comment by rsyring** [comment]
* I'd like to see some tests for this.  I know Keg Elements has lacked in this area, but no reason to persist my sloppyness.
* @nZac this also seems like a good time to get this running in CircleCI.  
* I'm not sure this field should be always required.  We want this field to be a boolean radio field. Whether or not a value is required should be an option.

File: keg_elements/forms/__init__.py [hunk 5]
```diff
@@ -200,6 +201,26 @@ def selected_choice_label(self):
         return value_dict.get(self.data)
 
 
+class RequiredRadioBooleanField(wtforms.fields.RadioField):
```
Comment (rsyring) [inline keg_elements/forms/__init__.py]: See comments in PR.  I think this should be named `BooleanRadioField` or `BoolRadioField`.

File: requirements/runtime.txt (hunk 1 above)
Comment (3noch) [inline requirements/runtime.txt]: KE already had [a direct dependency on this](https://github.com/level12/keg-elements/blob/master/keg_elements/forms/__init__.py#L17). I'll fix the name.

File: keg_elements/forms/__init__.py (hunk 2 above)
Comment (3noch) [inline keg_elements/forms/__init__.py]: No problem.

File: keg_elements/forms/__init__.py (hunk 3 above)
Comment (3noch) [inline keg_elements/forms/__init__.py]: I have a hard time thinking of other ways to validate True/False.

Comment (3noch) [inline keg_elements/forms/__init__.py]: Likewise, I have a hard time thinking of other ways to coerce True/False. Can you think of a legit case where you would want to coerce True/False to something other than a bool? We can always change the code later.

File: keg_elements/forms/__init__.py (hunk 4 above)
Comment (3noch) [inline keg_elements/forms/__init__.py]: Good catch.

**Comment by 3noch** [comment]
I intentionally made the use case for this field very specific. If you want more use cases, I can add support for them. In particular, I struggle to think of a case where you would want to use a `RadioField` for boolean data that is not required. Once you select an option in the radio field, you cannot de-select it. So it's only optional the very first time you use the form, after that it's required.

File: keg_elements/forms/__init__.py (hunk 3 above)
Comment (3noch) [inline keg_elements/forms/__init__.py]: Bear in mind that if you want to support coercions to non-bool types, then the user will also need to supply `choices`. At that point, the user of this class will be supplying almost every field manually and they might as well just use an actual `RadioField`.

**Comment by 3noch** [comment]
Also bear in mind that if "optionality" is parameterized, I will need to either default to "required" or create another subclass which does. The reason is because the form generator returns a `class` and not an instance, and thus does not have direct control over the constructor parameters. If it's possible to return a class factory instead, then that could work.

File: keg_elements/forms/__init__.py (hunk 3 above)
Comment (rsyring) [inline keg_elements/forms/__init__.py]: What if my DB backend had bit fields (mssql) that I wanted to use this with?  So the values actually need to be 0 or 1.

Validation: I could want to add a custom validator that makes sure that verifies a value on a different field.  If that field is X, this field can't be Y, etc.

The point is, the kwargs are there for a reason.  Why prevent them from being used when the code to add to them is almost as easy and leaves the API the most flexible?

Comment (3noch) [inline keg_elements/forms/__init__.py]: Because if you need that kind of customization there is a handy `RadioField` class ready for you. :)

Comment (rsyring) [inline keg_elements/forms/__init__.py]: My DB backend was a bad example b/c that's not what coerce does.  But yes, I could think of different reasons for wanting a different coerce method.  The end result may be True/False, but what if I also want 'true'/'false' or '0'/'1' or 'yes'/'no' to be coercable?  With this code, I have no options to make that happen.

Comment (3noch) [inline keg_elements/forms/__init__.py]: I'll add support for more validators. I need to think through the implications of supporting different semantic boolean types.

**Comment by rsyring** [comment]
> I struggle to think of a case where you would want to use a RadioField for boolean data that is not required. Once you select an option in the radio field, you cannot de-select it. So it's only optional the very first time you use the form, after that it's required.

Ok...that is a good point I hadn't considered.  I guess you can leave it as required.

File: keg_elements/forms/__init__.py (hunk 3 above)
Comment (3noch) [inline keg_elements/forms/__init__.py]: Isn't the serialization system for the DB backend separate from the Python type? If your DB uses 1/0, wouldn't SQLAlchemy hide that from you anyway and present a True/False type?

**Comment by 3noch** [comment]
@rsyring I am under the impression at this point that I'm doing this "properly" because it roughly follows the same pattern that your `coerce` function does for `SelectField` (i.e. it converts from a representation type to a semantic type; `str -> int` in that case).

**Comment by rsyring** [comment]
Hope this is right...we can fix later if needed.  :o/

**Pull request merged**